pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -140,8 +141,25 @@ public synchronized void shutDown() {
   @Override
   public InstanceResponseBlock execute(ServerQueryRequest queryRequest, ExecutorService executorService,
       @Nullable ResultsBlockStreamer streamer) {
+    MdcExecutor mdcExecutor = new MdcExecutor(executorService) {
+      @Override
+      protected boolean alreadyRegistered() {
+        return LoggerConstants.QUERY_ID_KEY.isRegistered();
+      }
+
+      @Override
+      protected void registerOnMDC() {
+        queryRequest.registerOnMdc();
```

Comment:
Why go through `queryRequest` to register on MDC ? Instead a call can be made directly to `LoggerConstants.QUERY_ID_KEY` ?

Reply:
To keep MDC management clean and centralized.